graphix/pattern.py

```diff
@@ -43,14 +43,15 @@ class Pattern:
         total number of nodes in the resource state
     """
 
-    def __init__(self, width=0, output_nodes=[]):
+    def __init__(self, width=0, input_nodes=[], output_nodes=[]):
```

Comment:
I think we can get rid of width argument here. maybe we can get rid of Nnode attribute too??

Reply:
As you say, the __init__ procedure of the Pattern class seems somewhat redundant. I think it can be more concise along with changes to the pattern generation code. I will try that.

Reply:
The dependency relation of attributes is (seq→output_nodes, width, Nnode), (output_nodes→width), input_nodes. (The right side of the arrow is determined if the left side is given.) 
We can use this to change the implementation so that the attributes assignment would be more consistent and concise for various types of pattern generation, but I think it is better to do this in other pull requests since there will be many changes.

Reply:
ok sounds good.